### ISSUE TITLE
Problematic use of implicit boolean conversion in episode 05

### ISSUE THREAD

**Issue by colinmorris** [datacarpentry/python-ecology-lesson#228]
This code works because `None` is "Falsey" - i.e. implicitly converted to False. But this isn't really explained.

```python
    def yearly_data_arg_test(all_data, start_year = None, end_year = None):
        """
        Modified from yearly_data_csv_writer to test default argument values!

        start_year --- the first year of data we want --- default: None - check all_data
        end_year --- the last year of data we want --- default: None - check all_data
        all_data --- DataFrame with multi-year data
        """

        if not start_year:
            start_year = min(all_data.year)
        if not end_year:
            end_year = max(all_data.year)
```
Also, theoretically this *could* cause buggy behaviour if someone specifies 0 (which is also Falsey) as the start or end year. (Unlikely, but maybe if you're working with a dataset of ancient documents or something?)

I'd suggest replacing the conditions with something more explicit. I don't have a strong opinion on whether it should `if start_year == None` or `if start_year is None`. The latter is the 'Pythonic' way but requires teaching a new keyword (and if someone asks why we use it instead of `==`, the answer is really esoteric).

**Comment by tiagopereira** [comment]
I think the approach `if start_year is None:` makes more sense. Agree that assuming `None` turns to `False` is not ideal.

**Comment by wrightaprilm** [comment]
Pushed fix in 7b823aa687d09.
